### DISCUSSION TITLE
Nested Group LDAP Auth not working in Airflow FAB

### DISCUSSION THREAD

**Discussion by ranga2crazyy** [apache/airflow#30710] [Q&A]
Hi All,

I am currently using flask_appbuilder.security.manager in order to provide LDAP authentication for my Airflow Users. 
While doing the AUTH_ROLES_MAPPING I have noticed that it only works for direct members of the AD groups not the nested groups. 
Has anyone been able to get this to work for nested groups?

Example:

in my current set up

_> AUTH_ROLES_MAPPING = {"CN=DIRECTGROUP,OU=!!sample,OU=!!sample,OU=!!sample,DC=sample,DC=sample" : ["Viewer", "executer"],
> 
> User A: member of DIRECTGROUP
> User B: member of NESTEDGROUP which is a member of DIRECTGROUP_

`Only User A would be able to login to my airflow instance as Viewer/Executer.`
`User B gets assigned to default  "Public" role since flask is not able to drill down to the subtree.`

I have tried using the Microsoft special string via AUTH_LDAP_SEARCH_FILTER but no luck. I browsed to different options in FAB but couldn't find one which supports this.

Current config options used(webserver_config.py) in Airflow 2.3.2

> import os
> from airflow.www.fab_security.manager import AUTH_LDAP
> basedir = os.path.abspath(os.path.dirname(__file__))
> 
> # Flask-WTF flag for CSRF
> WTF_CSRF_ENABLED = True
> 
> AUTH_LDAP_SEARCH = "dc=sample,dc=samp,dc=com" # the LDAP search base
> AUTH_LDAP_UID_FIELD = "sAMAccountName"
> AUTH_LDAP_BIND_USER & AUTH_LDAP_BIND_PASSWORD
> AUTH_USER_REGISTRATION = True
> AUTH_USER_REGISTRATION_ROLE = "Public" 
> AUTH_LDAP_GROUP_FIELD = "memberOf"
> AUTH_LDAP_SEARCH_FILTER='(&(objectClass=user)(memberOf:1.2.840.113556.1.4.1941:=CN=DIRECTGROUP,OU=!!sample,OU=!!sample,OU=!!sample,DC=sample,DC=sample))
> AUTH_ROLES_SYNC_AT_LOGIN=True
> PERMANENT_SESSION_LIFETIME=1800

has anyone able to get the nested LDAP group working via FAB?

**Comment by potiuk** [comment]
I think you should ask in FAB repo :). There is a better chance for answer there.

**Reply by lfrancke** [reply]
So it's easier for future people to find. 
This is the issue @ranga2crazyy created: https://github.com/dpgaspar/Flask-AppBuilder/issues/2022
There is a PR up but it hasn't seen any activtiy since November 2023.

@ranga2crazyy Did you find a solution for you?
